File: frontend/src/types/action.ts
export type ActionType =
  | "install"
  | "discard"
  | "rest"
  | "broadcast"
  | "followup"
  | "request_hint";

export interface BaseAction {
  actionId: string;
  actionType: ActionType;
  playerId: string;
  turn: number;
  timestamp: string;
}

export interface InstallAction extends BaseAction {
  actionType: "install";
  cardId: string;
  targetZone: string;
  targetSlot: number;
}

export interface DiscardAction extends BaseAction {
  actionType: "discard";
  cardId: string;
}

export interface RestAction extends BaseAction {
  actionType: "rest";
  hpRecovered: number;
}

export interface BroadcastAction extends BaseAction {
  actionType: "broadcast";
  message: string;
  targetAgentIds: string[] | "all";
}

export interface FollowupAction extends BaseAction {
  actionType: "followup";
  targetAgentId: string;
  question: string;
}

export type GameAction =
  | InstallAction
  | DiscardAction
  | RestAction
  | BroadcastAction
  | FollowupAction;

export interface ActionResult {
  success: boolean;
  actionType?: ActionType;
  message: string;
  hpDelta?: number;
  scoreDelta?: number;
  teamScoreDelta?: number;
  errorAdded?: boolean;
}

File: backend/app/schemas/action.py
from pydantic import BaseModel
from typing import Optional, List, Union, Dict

class GameActionRequest(BaseModel):
    playerId: str
    actionType: str # install, discard, rest, broadcast, followup
    cardId: Optional[str] = None
    targetZone: Optional[str] = None
    targetSlot: Optional[int] = None
    message: Optional[str] = None
    targetAgentIds: Optional[Union[str, List[str]]] = None
    question: Optional[str] = None
    targetAgentId: Optional[str] = None
    targetPlayerId: Optional[str] = None
    infoType: Optional[str] = None
    infoValue: Optional[str] = None
    targetCardIds: Optional[List[str]] = None

class ActionResponse(BaseModel):
    success: bool
    message: str
    data: Optional[Dict] = None
